Feature/tariik mvp/assign developers/ValidationToken

TarikVelic opened the pull request:
--------------------------------------------------------------------
 MAIN
 * Added employmentType (full-time, part-time)
 * Added token for validation if token exist or not

--------------------------------------------------------------------
 OTHER
 * <other msg> --------------------------------------------------------------------

TarikVelic committed:
employmentType, tokenValidation

--------------------------------------------------------------------
 MAIN
 * Added employmentType (full-time, part-time)
 * Added token for validation if token exist or not

--------------------------------------------------------------------
 OTHER
 * <other msg>
--------------------------------------------------------------------

TarikVelic committed:
employmentType, tokenValidation

--------------------------------------------------------------------
 MAIN
 * Added employmentType (full-time, part-time)
 * Added token for validation if token exist or not

--------------------------------------------------------------------
 OTHER
 * <other msg>
--------------------------------------------------------------------

becarselman commented on app.js:
```diff
@@ -37,6 +38,9 @@ app.get('/', (req, res) => {
   res.send('Hello World!');
 });
 
+// Ruta za provjeru valjanosti tokena
```
Please use only English

becarselman commented on controllers/validateToken.js:
```diff
@@ -0,0 +1,17 @@
+function validateToken(req, res) {
+  const { token } = req.body;
+
+  const isValid = tokenService.validateToken(token);
```
this need to be done in service.

becarselman commented on services/validateToken.js:
Idk is this pr wip or you didn't understand your task.

becarselman commented on services/validateToken.js:
the point of this task is to check if the jwt is valid or not. This will not help us in solving that problem.

TarikVelic committed:
Validation token controll

TarikVelic committed:
Validation token controll

becarselman commented on services/validateToken.js:
```diff
@@ -0,0 +1,16 @@
+
+function validateToken(token) {
+  try {
+    jwt.verify(token, secretKey);
```
we can reuse function from utils/helpers that we use in past. And also I this this function will return true and false so we don't need to create condition we just return what ever this function return. But test this

TarikVelic committed:
Review comment fix

TarikVelic committed:
Review comment fix

becarselman commented on controllers/validateToken.js:
```diff
@@ -0,0 +1,16 @@
+  const { token } = req.body;
+  const isValid = tokenService.validateToken(token);
+
+  if (isValid) {
```
this need to be done in service. I think validate token will return true and false so you don't need this condition

TarikVelic committed:
Review comments fix

TarikVelic committed:
Review comments fix

evlahovlja1 committed:
ElvaRivle/Feature/ProjectEditAndDelete

ElvaRivle/Feature/ProjectEditAndDelete

--------------------------------------------------------------------
 MAIN
 * added edit and delete routes for projects
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------

ElvaRivle committed:
ElvaRivle/Feature/ProjectEditAndDelete

--------------------------------------------------------------------
 MAIN
 * removed console.log
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------

AntMarkoLovric committed:
Merge pull request #26 from becarselman/ElvaRivle/Feature/ProjectEditAndDelete

ElvaRivle/Feature/ProjectEditAndDelete

TarikVelic committed:
Backend for developers avatar

TarikVelic committed:
Rebase with develop

ElvaRivle committed:
ElvaRivle/Misc/NewUsers

ElvaRivle/Misc/NewUsers

--------------------------------------------------------------------
 MAIN
 * nicer users that make more sense for login
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------

ElvaRivle committed:
ElvaRivle/Misc/LoginReturnData

--------------------------------------------------------------------
 MAIN
 * on login return name and surname as well
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * /

AntMarkoLovric committed:
Merge pull request #29 from becarselman/ElvaRivle/Misc/LoginReturnData

ElvaRivle/Misc/LoginReturnData

AntMarkoLovric committed:
Merge pull request #28 from becarselman/ElvaRivle/Misc/NewUsers

ElvaRivle/Misc/NewUsers

TarikVelic committed:
small code changes

TarikVelic committed:
Deleted employeesType

TarikVelic committed:
Deleted employeesType

TarikVelic committed:
rebase with develop

TarikVelic committed:
small code changes
